docs-starlight/src/content/docs/02-guides/01-terralith-to-terragrunt/05-step-2-refactoring.mdx

```diff
@@ -0,0 +1,687 @@
+  source = "../catalog/modules/ddb"
+
+  name = var.name
+}
+
+module "iam" {
+  source = "../catalog/modules/iam"
+
+  name = var.name
+
+  aws_region = var.aws_region
+
+  s3_bucket_arn      = module.s3.arn
+  dynamodb_table_arn = module.ddb.arn
+}
+
+module "lambda" {
+  source = "../catalog/modules/lambda"
+
+  name = var.name
+
+  aws_region = var.aws_region
+
+  s3_bucket_name      = module.s3.name
+  dynamodb_table_name = module.ddb.name
+  lambda_zip_file     = var.lambda_zip_file
+  lambda_role_arn     = module.iam.arn
+}
+```
+
+We also want to forward outputs from these modules into our root module so that we can access them from the `tofu` CLI.
+
+```hcl
+# live/outputs.tf
+
+output "lambda_function_url" {
+  description = "URL of the Lambda function"
+  value       = module.lambda.url
+}
+
+output "lambda_function_name" {
+  description = "Name of the Lambda function"
+  value       = module.lambda.name
+}
+
+output "s3_bucket_name" {
+  description = "Name of the S3 bucket for static assets"
+  value       = module.s3.name
+}
+
+output "s3_bucket_arn" {
+  description = "ARN of the S3 bucket for static assets"
+  value       = module.s3.arn
+}
+
+output "dynamodb_table_name" {
+  description = "Name of the DynamoDB table for asset metadata"
+  value       = module.ddb.name
+}
+
+output "dynamodb_table_arn" {
+  description = "ARN of the DynamoDB table for asset metadata"
+  value       = module.ddb.arn
+}
+
+output "lambda_role_arn" {
+  description = "ARN of the Lambda execution role"
+  value       = module.iam.arn
+}
+```
+
+We can also reduce the amount of content in the optional variables file, now that each of the modules define the variables that matter to them. This keeps the API of each module clean, as each module exposes the variables that specifically control them.
+
+```hcl
+# live/vars-optional.tf
+
+variable "aws_region" {
+  description = "AWS region for all resources"
+  type        = string
+  default     = "us-east-1"
+}
+
+variable "force_destroy" {
+  description = "Force destroy S3 buckets (only set to true for testing or cleanup of demo environments)"
+  type        = bool
+  default     = false
+}
+
+```
+
+After all this refactoring, we'll want to run a `plan` to make sure we can safely apply our changes.
+
+<Aside type="note">
+You'll need to re-initialize now that you're using modules here.
+</Aside>
+
+```bash
+$ tofu init
+$ tofu plan
+...
+
+Plan: 11 to add, 0 to change, 11 to destroy.
+...
+```
+
+Oh no! After all our refactors, we've introduced changes that would *completely destroy* all of the infrastructure we've created so far!
+
+Take a breath, and relax. This is a common scenario that you need to become comfortable with as you learn how to refactor and adjust IaC for scalability and maintainability. You leveraged the built-in protections of plans to give you a dry-run of your infrastructure updates, and can reason about why OpenTofu is trying to do what it's doing here to avoid catastrophe.
```

Comment:
```suggestion
This is a common scenario that you need to become comfortable with as you learn how to refactor and adjust IaC for scalability and maintainability. You leveraged the built-in protections of plans to give you a dry-run of your infrastructure updates, and can reason about why OpenTofu is trying to do what it's doing here to avoid catastrophe.
```